File: BackBarbearia/Models/Agenda.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace Salao.Models
{
    public class Agenda
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id{get;set;}
        public int BarbeiroId{get;set;}
        public Barbeiro? Barbeiro{get;set;}
        public List<AtendimentoAgendado>? Atendimentos{get;set;}
        public List<HorarioLivre>? Horarios{get;set;}
    }
}

File: FrontBarbearia/src/app/Agenda.ts
import { AtendimentoAgendado } from "./AtendimentoAgendado";
import { Barbeiro } from "./Barbeiro";
import { HorarioLivre } from "./HorarioLivre";


export class Agenda {
    id: number = 0;
    barbeiroId: number = 0;
    barbeiro: Barbeiro | undefined;
    atendimentos: Array<AtendimentoAgendado> | undefined;
    horarios : Array<Date> |undefined; 
}